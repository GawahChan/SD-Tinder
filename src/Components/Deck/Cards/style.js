import styled from 'styled-components';
import { motion } from 'framer-motion';

export const CardsContainer = styled.div`
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    min-height: 100vh;
`
// eslint-disable-next-line 
export const CardContainer = styled(motion.div).attrs(props => ({
    style: {
        boxShadow: props.actionColour === 'liked' ? '0px 0px 20px #15ff00' :
            props.actionColour === 'disliked' ? '0px 0px 20px #ff0000' :
                props.actionColour === 'superliked' ? '0 0 20px #02fdfd' : '0px 0px 10px #080808',

        border: props.actionColour === 'liked' ? '1px solid #15ff00' :
            props.actionColour === 'disliked' ? '1px solid #ff0000' :
                props.actionColour === 'superliked' ? '1px solid #02fdfd' : 'none'
    }
}))
    `
    position: absolute;
    border-radius: 10px;
    overflow: hidden;

    width: 320px;
    height: 500px;
        
    transition: box-shadow 0.5s, border 0.5s;
    will-change: transform;
`

export const CardContents = styled(motion.div)`
    width: 100%;
    height: 100%;

    display: flex;
    flex-direction: column;
    justify-content: flex-end; 

    background-image: ${props => {
        return (
            `linear-gradient(rgba(0,0,0,0), rgba(0,0,0,0), rgba(0,0,0,0.5), rgba(0,0,0,1)),
            url(${require(`../../../Common/Profile/${props.img}.jpg`)})`
        )

    }};
    background-position: top;
    background-size: cover;

    transition: box-shadow 0.5s;
    will-change: transform;
`

export const CardContent = styled.div`
    padding: 1rem;
    color: #DDDDDD;
`
// eslint-disable-next-line 
export const CardOverlay = styled.h1.attrs(props => ({
    style: {
        color: props.actionColour === 'LIKED' ? '#15ff00' :
            props.actionColour === 'DISLIKED' ? '#ff0000' :
                props.actionColour === 'SUPERLIKED' ? '#02fdfd' : 'black'
    }
}))
    `
    height: 30%;
    width: 100%;

    display: flex;
    justify-content: center;
    align-items: center;

    font-size: 3rem;
    font-weight: 500;

`
export const CardTitle = styled.h1`
    margin: 0;
    font-weight: 400;
    font-size: 2rem;
`

export const CardSubtitle = styled.div`
    display: flex;
    align-items: center;
    justify-content: space-between;
`

export const CardInfoButton = styled(motion.div)`
    background-image: url(${require(`../../../Common/Icons/info.svg`)});
    background-size: cover;

    height: 25px;
    width: 25px;
    cursor: pointer;
    outline: 0;
`

export const CardButtonContainer = styled.div`
    display: flex;
    align-items: center;
    justify-content: center;
    margin: 5px; 
`

export const CardLikedButton = styled(motion.div)`
    background-image: url(${require(`../../../Common/Icons/liked.svg`)});
    background-size: cover;
    height: 50px;
    width: 50px;
    margin: 10px;
    cursor: pointer;
    outline: 0;
`

export const CardDislikedButton = styled(CardLikedButton)`
    background-image: url(${require(`../../../Common/Icons/disliked.svg`)});

`

export const CardSuperlikedButton = styled(CardLikedButton)`
    background-image: url(${require(`../../../Common/Icons/superliked.svg`)});

`