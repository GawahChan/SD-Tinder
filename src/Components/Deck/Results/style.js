import styled from 'styled-components';

export const ResultsContainer = styled.div`
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;

    min-height: 100vh;

    text-align: center;
    margin: 30px;
`

export const ResultDisplay = styled.div`
    display: flex;
    flex-wrap: wrap;
    justify-content: space-evenly;
`

export const ResultHeader = styled.h1`
    color: white;
    font-weight: 400;
    font-size: 2.5rem;
`

export const ResultTitle = styled(ResultHeader)`
    font-size: 1.5rem;
`

export const ResultSubTitle = styled(ResultHeader)`
    font-size: 1rem;
    margin: 0;
`

export const ResultProfile = styled.div`
    background-image: ${props => {
        return (
            `linear-gradient(rgba(0,0,0,0), rgba(0,0,0,0), rgba(0,0,0,0.6), rgba(0,0,0,1)),
            url(${require(`../../../Common/Profile/${props.img}.jpg`)})     `
        )
    }};
    
    border-radius: 10px;
    box-shadow: 1px 1px 8px #333;
    margin: 10px;

    background-size: cover;
    background-position: top;
    overflow: hidden;

    display: flex;
    flex-direction: column;
    justify-content: flex-end;
    
    min-width: 250px;
    min-height: 250px;
`