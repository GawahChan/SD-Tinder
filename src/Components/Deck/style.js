import styled from "styled-components";
import { motion } from "framer-motion";

export const CardsContainer = styled(motion.div)`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  min-height: 100vh;
  flex: 1;
`;
// eslint-disable-next-line
export const CardContainer = styled(motion.div).attrs(props => ({
  style: {
    boxShadow:
      props.actionColour === "liked"
        ? "0px 0px 20px #15ff00"
        : props.actionColour === "disliked"
        ? "0px 0px 20px #ff0000"
        : props.actionColour === "superliked"
        ? "0 0 20px #02fdfd"
        : "0px 0px 10px #080808",

    border:
      props.actionColour === "liked"
        ? "1px solid #15ff00"
        : props.actionColour === "disliked"
        ? "1px solid #ff0000"
        : props.actionColour === "superliked"
        ? "1px solid #02fdfd"
        : "none"
  }
}))`
  position: absolute;
  border-radius: 10px;
  overflow: hidden;
  cursor: pointer;

  width: 320px;
  height: 480px;

  transition: box-shadow 0.5s, border 0.5s;
  will-change: transform;
`;

export const CardContents = styled(motion.div)`
  width: 100%;
  height: 100%;

  display: flex;
  flex-direction: column;
  justify-content: flex-end;

  background-image: ${props =>
    `url(${require(`../../Common/Profile/${props.img}.jpg`)})`};
  background-position: top;
  background-size: cover;

  transition: box-shadow 0.5s;
  will-change: transform;
`;

export const CardContent = styled.div`
  padding: 10px;
  box-sizing: border-box;
  color: #dddddd;
  background: linear-gradient(
    rgba(0, 0, 0, 0),
    rgba(0, 0, 0, 0.7),
    rgba(0, 0, 0, 0.8),
    rgba(0, 0, 0, 1)
  );
`;

// eslint-disable-next-line
export const CardOverlay = styled.h1.attrs(props => ({
  style: {
    color:
      props.actionColour === "LIKED"
        ? "#15ff00"
        : props.actionColour === "DISLIKED"
        ? "#ff0000"
        : props.actionColour === "SUPERLIKED"
        ? "#02fdfd"
        : "black"
  }
}))`
  height: 30%;
  width: 100%;

  display: flex;
  justify-content: center;
  align-items: center;

  font-size: 3rem;
  font-weight: 500;
`;

export const CardSubtitleContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  box-sizing: border-box;
  padding-top: 5px;
  padding-bottom: 5px;
`;

export const CardButtonContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
`;

export const Container = styled.div`
  box-sizing: border-box;
  padding: 5px;
`;
