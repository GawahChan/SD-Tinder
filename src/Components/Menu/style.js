import styled, { css } from "styled-components";
import { Link } from "react-router-dom";
import { motion } from "framer-motion";

export const MenuContainer = styled.div`
  position: fixed;
  z-index: 1;
  background-color: none;

  ${props =>
    props.displayMenu &&
    css`
      background-color: rgba(250, 250, 250, 0.5);
      height: 100%;
      width: 100%;
    `}
`;

export const MenuPanelContainer = styled(motion.div)`
  position: absolute;
  height: 100vh;
  width: 320px;

  background-color: rgba(250, 250, 250, 0.9);
  left: -350px;

  ${props =>
    props.displayMenu &&
    css`
      left: 0px;
    `}

  transition: left 1s;
`;

export const MenuPanel = styled.div`
  box-sizing: border-box;

  padding: 75px 5px 10px 5px;
`;

export const MenuLink = styled(Link)`
  text-decoration: none;
`;

export const Container = styled.div`
  box-sizing: border-box;
  padding: 5px;
`;
