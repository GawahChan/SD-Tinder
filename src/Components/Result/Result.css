.result {
    text-align: center;
    margin: 30px 0 30px 0;
}

.result h1 {
    color: white;
    font-weight: 400;
    font-size: 2.5rem;
}

.result h2 {
    color: white;
    font-weight: 400;
    font-size: 1.5rem;
}

.result h3 {
    color: white;
    font-weight: 400;
    font-size: 1rem;
    margin: 0;
}

.result-profiles {
    display: flex;
    flex-wrap: wrap;
    justify-content: space-evenly;
}