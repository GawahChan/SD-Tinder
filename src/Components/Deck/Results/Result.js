import React from "react";
import ResultCard from "./ResultCard";
import { Title, Subtitle } from "../../../Common/Styled/Typography.styles";
import { ResultDisplay, Container } from "./style";

function Result(props) {
  return (
    <Container>
      <Title>{props.title}</Title>
      <ResultDisplay>
        {props.stuntDouble.length > 0 ? (
          props.stuntDouble.map(stuntDouble => {
            return (
              <ResultCard
                key={stuntDouble.id}
                img={stuntDouble.img}
                name={stuntDouble.name}
              />
            );
          })
        ) : (
          <Subtitle>{props.subTitle}</Subtitle>
        )}
      </ResultDisplay>
    </Container>
  );
}

export default Result;
