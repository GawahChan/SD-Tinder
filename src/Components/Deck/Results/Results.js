import React from "react";
import PropTypes from "prop-types";
import Result from "./Result";
import { Header } from "../../../Common/Styled/Typography.styles";
import { ResultsContainer } from "./style";

function Results({ superliked, liked, disliked }) {
  return (
    <ResultsContainer>
      <Header>There's no one new around you</Header>
      <Result
        title={"People you superliked!"}
        stuntDouble={superliked}
        subTitle={"You didn't superlike anyone"}
      />
      <Result
        title={"People you liked... let's hope they like you too!"}
        stuntDouble={liked}
        subTitle={"You didn't like anyone"}
      />
      <Result
        title={"People you disliked"}
        stuntDouble={disliked}
        subTitle={"Cast a wide net they said..."}
      />
    </ResultsContainer>
  );
}

Results.propTypes = {
  superliked: PropTypes.array,
  liked: PropTypes.array,
  disliked: PropTypes.array
};

export default Results;
