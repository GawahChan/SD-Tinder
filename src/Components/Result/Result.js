import React from 'react';
import ResultCard from './ResultCard.js';
import './Result.css';

function Result(props) {
    return (
        <div className='result'>
            <h1>There's no one new around you</h1>
            <div className='result-superliked'>
                <h2>People you Superliked!</h2>
                <div className='result-profiles superliked'>
                    {
                        props.superliked.length > 0 ?
                            props.superliked.map(stuntDouble => {
                                return (
                                    <ResultCard
                                        key={stuntDouble.id}
                                        img={stuntDouble.img}
                                        name={stuntDouble.name}
                                    />
                                )
                            }) :
                            <h3>You didn't superlike any profiles</h3>
                    }
                </div>
            </div>
            <div className='result-liked'>
                <h2>People you liked... let's hope they like you too!</h2>
                <div className='result-profiles liked'>
                    {
                        props.liked.length > 0 ?
                            props.liked.map(stuntDouble => {
                                return (
                                    <ResultCard
                                        key={stuntDouble.id}
                                        img={stuntDouble.img}
                                        name={stuntDouble.name}
                                    />
                                )
                            }) :
                            <h3>You didn't like any profiles</h3>
                    }
                </div>
            </div>
            <div className='result-disliked'>
                <h2>People you disliked</h2>
                <div className='result-profiles disliked'>
                    {
                        props.disliked.length > 0 ?
                            props.disliked.map(stuntDouble => {
                                return (
                                    <ResultCard
                                        key={stuntDouble.id}
                                        img={stuntDouble.img}
                                        name={stuntDouble.name}
                                    />
                                )
                            }) :
                            <h3>Cast a wide net they said...</h3>
                    }
                </div>
            </div >
        </div>
    );
}

export default Result;