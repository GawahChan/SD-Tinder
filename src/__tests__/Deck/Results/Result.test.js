import React from 'react';
import { shallow } from 'enzyme';
import Result from '../../../Components/Deck/Results/Result';
import { ResultSubTitle } from '../../../Components/Deck/Results/style';

import Data from '../../../Data';

describe('Result Component', () => {
    let data = [Data[0], Data[1], Data[2]];

    let props = { stuntDouble: data, title: 'Title', subTitle: 'subtitle' }
    let result = shallow(<Result  {...props} />)

    it('renders correctly', () => {
        expect(result).toMatchSnapshot();
    });

    it('renders in <ResultCard /> 3 times due to props above', () => {
        expect(result.find('ResultCard').length).toEqual(3);
    });

    it('does not render <ResultSubTitle />', () => {
        expect(result.find(ResultSubTitle).length).toEqual(0);
    });

    describe('When there are no props in stuntDouble', () => {
        let props = { stuntDouble: [], title: 'Title', subTitle: 'subTitle' }
        let result = shallow(<Result {...props} />)

        it('Does not render <ResultCard />', () => {
            expect(result.find('ResultCard').length).toEqual(0);
        });

        it('Renders the <ResultSubTitle /> instead', () => {
            expect(result.find(ResultSubTitle).length).toEqual(1);
        });
    });
});