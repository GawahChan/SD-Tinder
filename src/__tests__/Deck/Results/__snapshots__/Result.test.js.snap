// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result Component renders correctly 1`] = `
<div>
  <Styled(styled.h1)>
    Title
  </Styled(styled.h1)>
  <styled.div>
    <ResultCard
      img="Alicia-Vela-Bailey"
      key="0"
      name="Alicia Vela-Bailey"
    />
    <ResultCard
      img="Richard-Cetrone"
      key="1"
      name="Richard Cetrone"
    />
    <ResultCard
      img="Renae-Moneymaker"
      key="2"
      name="Renae Moneymaker"
    />
  </styled.div>
</div>
`;
